telemetry/metrics.go

```diff
@@ -14,6 +15,21 @@ import (
 	"github.com/prometheus/common/expfmt"
 )
 
+// globalTelemetryEnabled is a private variable that stores the telemetry enabled state.
+// It is set on initialization and does not change for the lifetime of the program.
+var globalTelemetryEnabled atomic.Bool
+
+// initTelemetry sets the global variable based on the configuration.
+// It is called only once, at startup, to set the telemetry enabled state.
+func initTelemetry(enabled bool) {
+	globalTelemetryEnabled.Store(enabled)
+}
+
+// isTelemetryEnabled provides controlled access to check if telemetry is enabled.
+func isTelemetryEnabled() bool {
```

Comment:
personal preference: with a standard bool, this method can be inlined. If you want to provide the information to other packages or modules, it can make sense to make this public though.